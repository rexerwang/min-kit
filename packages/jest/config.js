/**
 * @typedef {import('jest').Config Config}
 */

const defineJestConfig = require('./config/taro-jest/index').default

/** @type {Config} */
const baseConfig = defineJestConfig({
  globalSetup: require.resolve('./config/globalSetup.js'),
  setupFilesAfterEnv: [require.resolve('./config/setupAfterEnv.js')],
  testEnvironment: '@happy-dom/jest-environment', // 'jsdom'
  testMatch: ['<rootDir>/src/__tests__/**/*.test.ts?(x)'],
  collectCoverageFrom: ['<rootDir>/src/**/*.ts?(x)', '!**/__tests__/**'],
  coverageReporters: process.env.CI ? ['json'] : ['html', 'text'],
  passWithNoTests: true,
})

/**
 * defineJestConfig
 * @param {Partial<Config>} [config]
 * @returns {Config}
 */
module.exports = (config = {}) => {
  let { setupFilesAfterEnv, collectCoverageFrom, ...configs } = baseConfig

  if (Array.isArray(config.setupFilesAfterEnv)) {
    setupFilesAfterEnv = setupFilesAfterEnv.concat(config.setupFilesAfterEnv)
  }

  if (Array.isArray(config.collectCoverageFrom)) {
    collectCoverageFrom = [collectCoverageFrom[0], ...config.collectCoverageFrom, collectCoverageFrom[1]]
  }

  return {
    ...configs,
    ...config,
    setupFilesAfterEnv,
    collectCoverageFrom,
  }
}
