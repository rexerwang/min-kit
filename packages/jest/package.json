{
  "name": "@min-kit/jest",
  "version": "0.5.11",
  "private": true,
  "description": "jest config & utils for internal",
  "author": "rexerwang",
  "license": "MIT",
  "repository": "rexerwang/min-kit",
  "exports": {
    ".": "./utils.ts",
    "./config": "./config.js"
  },
  "devDependencies": {
    "@happy-dom/jest-environment": "^14.7.1",
    "@tarojs/test-utils-react": "^0.1.1",
    "@testing-library/jest-dom": "^6.4.8",
    "@testing-library/react": "^16.0.1"
  }
}
