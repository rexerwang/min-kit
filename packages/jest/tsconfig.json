{
  "extends": "../../tsconfig.root.json",
  "include": ["utils.ts"]
}
