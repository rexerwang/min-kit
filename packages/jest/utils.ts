import TestUtils from '@tarojs/test-utils-react'

export const createTaroTestUtils = () => {
  const testUtils = new TestUtils()
  return Object.assign(testUtils, { unmount: testUtils.unmout })
}

export const spyOnConsole = (methods = ['error', 'warn', 'info', 'debug']) =>
  methods.map((method: any) => {
    const spy = jest.spyOn(console, method)
    spy.mockImplementation(jest.fn())
    return spy
  })

export * from '@testing-library/react'
