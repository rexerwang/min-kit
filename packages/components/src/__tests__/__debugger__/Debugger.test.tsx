// import * as _extends from '@min-kit/extends'
import { act, fireEvent, render, screen } from '@min-kit/jest'

import Debugger from '../../debugger/ui/debugger'

describe('<Debugger />', () => {
  it('should render <Debugger /> toMatchSnapshot', () => {
    expect(render(<Debugger />).asFragment()).toMatchSnapshot()
  })

  it('should toggle when click button', () => {
    const { container } = render(<Debugger />)
    const panel = container.querySelector('.min-debug-panel')!.parentElement!
    const button = screen.getByText('debugger')

    // initial
    expect(panel.classList.contains('hidden')).toBeTruthy()
    expect(button.classList.contains('hidden')).toBeFalsy()

    // panel visible after click debugger
    act(() => {
      fireEvent.click(button)
    })
    expect(panel.classList.contains('hidden')).toBeFalsy()
    expect(button.classList.contains('hidden')).toBeTruthy()

    // panel hidden after click mask
    act(() => {
      fireEvent.click(panel.querySelector('.mask')!)
    })
    expect(panel.classList.contains('hidden')).toBeTruthy()
    expect(button.classList.contains('hidden')).toBeFalsy()
  })

  it('should dispatch onMove event correctly', () => {
    const onMove = jest.fn()
    render(<Debugger onMove={onMove} />)
    const button = screen.getByText('debugger')
    act(() => {
      const event = new CustomEvent('change', { detail: { x: 1, y: 1 } })
      fireEvent(button, event)
    })
    expect(onMove).toHaveBeenCalledWith({ x: 1, y: 1 })
  })
})
