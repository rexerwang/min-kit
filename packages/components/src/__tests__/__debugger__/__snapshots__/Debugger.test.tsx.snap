// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Debugger /> should render <Debugger /> toMatchSnapshot 1`] = `
<DocumentFragment>
  <taro-view-core
    class="min-debugger"
  >
    <!---->
    <taro-view-core
      class="hidden"
    >
      <!---->
      <taro-view-core
        class="min-debug-panel"
      >
        <!---->
        <taro-view-core
          class="mask"
        >
          <!---->
        </taro-view-core>
        <taro-view-core
          class="container"
          style="top: 80px; bottom: 34px;"
        >
          <!---->
          <taro-view-core
            class="tabs"
          >
            <!---->
            <taro-view-core
              class="tabHeader"
            >
              <!---->
              <taro-view-core
                class="tab active"
              >
                <!---->
                请求
              </taro-view-core>
              <taro-view-core
                class="tab"
              >
                <!---->
                应用
              </taro-view-core>
              <taro-view-core
                class="tab"
              >
                <!---->
                状态
              </taro-view-core>
            </taro-view-core>
            <taro-view-core
              class="tabBody"
            >
              <!---->
              <taro-view-core
                class="NetworkPanel panel"
              >
                <!---->
                <taro-scroll-view-core
                  class="main"
                  scroll-y="true"
                >
                  <!---->
                  <taro-view-core
                    class="empty"
                  >
                    <!---->
                    Recording network activity...
                  </taro-view-core>
                </taro-scroll-view-core>
                <taro-view-core
                  class="footer"
                >
                  <!---->
                  <taro-button-core
                    class=""
                    plain="true"
                  >
                    <!---->
                    筛选
                  </taro-button-core>
                  <taro-button-core
                    class=""
                    plain="true"
                  >
                    <!---->
                    清空
                  </taro-button-core>
                </taro-view-core>
              </taro-view-core>
            </taro-view-core>
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
    <taro-movable-area-core
      class="movable-area"
      style="top: 0px; bottom: 34px;"
    >
      <taro-movable-view-core
        animation="false"
        class="movable-view"
        direction="all"
        style="transform: translateX(0px) translateY(0px) translateZ(0px) scale(1);"
        x="0"
        y="0"
      >
        debugger
      </taro-movable-view-core>
    </taro-movable-area-core>
  </taro-view-core>
</DocumentFragment>
`;
